### PR TITLE
feat: Print the output of an invoked script to output

### LINKED ISSUE #330: Invoking a Script in VSCode should return the results
Currently, it only returns pass/fail

### PR DESCRIPTION
**Pull request by Marwes**
Is this how we want to display it? There seems to be a table view that is opened when queries are run, should this use that as well?

What is the difference between scripts and queries? (Why couldn't this use the API for running "Queries"?)

Closes #330

### CODE REVIEW

**Review by rockstar — CHANGES_REQUESTED** [review]
> What is the difference between scripts and queries? (Why couldn't this use the API for running "Queries"?)

Scripts are entities that live on the server. We _could_ (and considered) just getting the contents of the script and executing it in the same code path, but ultimately, we wanted to fix the issue preventing us from executing it (a client API issue). If we use the "invoke" API, there's a execution history that gets updated with success/failure and other metadata, similar to tasks. We wanted to retain that.

I think this work is on the right track, and wish that we could just output this to the output channel but, historically, our query results show up in a table. I have a plan for changing that, but the user experience will suffer if we depart from it so abruptly.

File: src/views/TreeView.ts [hunk 1]
```diff
@@ -496,10 +498,31 @@ export class Script extends vscode.TreeItem {
...
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            const _response = await scriptsApi.postScriptsIDInvoke({ scriptID: this.script.id!, body: {} })
+            if (this.output) {
+                this.output.clear()
+            }
+            const self = this // eslint-disable-line @typescript-eslint/no-this-alias
+            let first = true
+            await new Promise<void>((resolve, reject) => {
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                scriptsApi.invoke(this.script.id!).consumeRows({
```
[the remark this answers is not in the document]
Comment (Marwes) [reply src/views/TreeView.ts]: Done.

File: src/models.ts [hunk 2]
```diff
@@ -32,38 +32,48 @@ export class QueryResult {
...
-                        result.push(currentTableResult)
-                    }
-                    resolve(result)
-                }
-            })
+            client.queryRows(query, queryResultObserver(resolve, reject))
+        })
+    }
+
+    static async parseCSVResponse(response : AnnotatedCSVResponse) : Promise<QueryResult> {
```
Comment (rockstar) [inline src/models.ts:40]: Hm, I don't quite understand the need for this. The intent of the upstream changes to the client apis was that it had the same api as the run `queryRows` api, so would technically only need to use the `queryResultObserver` - if that's not the case, I think we need to file a bug.

If we can't get that same API, I think this function (a) should move to where it's used, and (b) should have a comment with the link to the bug about having an inconsistent API.

File: src/models.ts [hunk 3]
```diff
@@ -32,38 +32,48 @@ export class QueryResult {
...
+        return new Promise((resolve, reject) => {
+            response.consumeRows(queryResultObserver(resolve, reject))
         })
     }
 }
+
+function queryResultObserver(resolve : (result : QueryResult | PromiseLike<QueryResult>) => void, reject : (reason ?: Error) => void) : FluxResultObserver<string[]> {
```
Comment (rockstar) [inline src/models.ts:47]: This is a good refactor.

**Review by rockstar — APPROVED** [review]
Couple of questions regarding our client apis, and maybe a request to move some code and add a comment. I leave the merge to your discretion.

File: package.json [hunk 4]
```diff
@@ -289,6 +289,11 @@
           "when": "view == influxdb && viewItem == script",
           "group": "influxdb@1"
         },
+        {
```
[the remark this answers is not in the document]
Comment (Marwes) [reply package.json:292]: Yes, this just assigns the command to the right click menu.

File: src/models.ts (hunk 2 above)
Comment (Marwes) [reply src/models.ts]: Query has `client.queryRows(query, queryResultObserver(resolve, reject))` and invoke uses `client.invoke(id).consumeRows(queryResultObserver(resolve, reject))`.

Do you mean that it should be `client.invoke(id, queryResultObserver(resolve, reject))`? I tweaked it a bit in the latest https://github.com/influxdata/vsflux/pull/359/commits/564322f67e234bf39410c421e6a52180c2262254

**Comment by Marwes** [comment]
@rockstar `influxdb-client-apis` seems to have broken semver (noticed due to CI breakage), locked the dependency down and removed `orgID` as suggested in https://github.com/influxdata/influxdb-client-js/issues/404 .

**Comment by Marwes** [comment]
We may want to change the `npm i` call on CI to `npm ci` instead so we use the dependencies in the lockfile instead of automatically updating it.

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (rockstar) [reply package.json:298]: Ah, I see, it's already got the entry banning it from the command palette.

File: src/models.ts (hunk 2 above)
Comment (rockstar) [reply src/models.ts]: Hm, that feels inconsistent, but nothing we can do about it, I guess.

**Pull request merged**